scala-compile.el

```diff
@@ -0,0 +1,116 @@
+;;; scala-compile.el --- batch compile scala -*- lexical-binding: t -*-
+
+;; Copyright (C) 2020 Sam Halliday
+;; License: GPL 3 or any later version
+
+;;; Commentary:
+;;
+;;  An idiomatic `compilation-mode' batch compilation command that detects
+;;  warnings and errors, extracting line numbers, columns and ranges.
+;;
+;;  Relies on a batch tool, such as the sbt thin client (`sbt --client`) which
+;;  can be compiled to native binary (`sbtn`) from their repo with `sbt
+;;  buildNativeThinClient` or mystery meat binaries downloaded from
+;;  https://github.com/sbt/sbtn-dist/releases/
+;;
+;;; Code:
+
+(require 'compile)
+(require 'ansi-color)
+(require 'files)
+(require 'subr-x)
+
+;; TODO flycheck integration https://emacs.stackexchange.com/questions/51894
+;; TODO is -Dsbt.supershell=false needed?
+
+(defcustom scala-compile-always-ask t
+  "`scala-compile' will remember the last command for the buffer unless set."
+  :type 'booleanp
+  :group 'scala)
+
+(defcustom scala-compile-suggestion nil
+  "Files can specify a suggested command to run, e.g. runMain and testOnly."
+  :type 'stringp
+  :group 'scala
+  :safe 'stringp
+  :local t)
+
+;; the geniuses at EPFL completed changed the output format between Scala 2 and
```

Comment:
While I do understand where you are coming with this, can we please have a more neutral comment in the PR?